### PR TITLE
Improvement in search modules

### PR DESCRIPTION
**Pull request by ktg-123**
I have made the changes as suggested first in the Reddit module; If they seem good, I will update other modules within this PR only. I will wait for the review before pushing other updates, If any changes are suggested, then I will do them in other modules as well.

### CODE REVIEW

File: modules/search/reddit.py [hunk 1]
```diff
@@ -31,71 +31,69 @@ class Module(BaseModule):
...
 		limit = self.options['limit']
 		count = self.options['count']
 		engine = self.options['engine'].split(',')
-		google_q = f"site:www.reddit.com inurl:{query}"
-		q = f"site:www.reddit.com {query}"
-		yippy_q = f'"www.reddit.com" {query}'
-		millionshort_q = f'site:www.reddit.com "{query}"'
-		qwant_q = f'site:www.reddit.com {query}'
+		q_format = {
+			'_q': f"site:www.reddit.com {query}",
```
Comment (saeeddhqan) [inline modules/search/reddit.py]: default_q is better than _q.
q_formats is better than...

File: modules/search/reddit.py [hunk 2]
```diff
@@ -31,71 +31,69 @@ class Module(BaseModule):
...
+		threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=thread_count)
+		futures = (threadpool.submit(
+			function, name, q, q_format, limit, count) for name in engines if name in self.meta['sources'])
+		for _ in concurrent.futures.as_completed(futures):
+			pass
+
+	def search(self, name, q, q_format, limit, count):
+		engine = getattr(self, name)
+		name = engine.__name__
+		query = name + '_q'
```
Comment (saeeddhqan) [inline modules/search/reddit.py]: don't use try/except for this such trivial task:
`query = q_formats[f"{name}_q"] if f"{name}_q" in q_formats else q_formats['default_q']`
q_formats variable name also is better

**Review by saeeddhqan — COMMENTED** [review]
Great.

File: modules/search/reddit.py (hunk 2 above)
Comment (ktg-123) [reply modules/search/reddit.py]: Thanks! will do the same

**Comment by ktg-123** [comment]
I have made the suggested changes. If they are good to go I shall push changes in other modules as well.

**Comment by saeeddhqan** [comment]
All good. continue.
Thank you

**Comment by ktg-123** [comment]
I have made the changes in the search modules. Please review

**Comment by saeeddhqan** [comment]
I'll check it out

**Pull request merged**